src/VisualStudio/Core/Impl/CodeModel/CodeModelState.cs

```diff
@@ -14,11 +14,13 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.CodeModel
     /// </summary>
     internal sealed class CodeModelState
     {
+        private readonly HostLanguageServices _languageServices;
+
         public IThreadingContext ThreadingContext { get; }
         public IServiceProvider ServiceProvider { get; }
-        public ICodeModelService CodeModelService { get; }
-        public ISyntaxFactsService SyntaxFactsService { get; }
-        public ICodeGenerationService CodeGenerator { get; }
+        public ICodeModelService CodeModelService => _languageServices.GetRequiredService<ICodeModelService>();
```

Comment:
Why the change? To reduce memory?

Reply:
This change is now removed from this pull request.